frontend/src/metabase/dashboard/reducers-typed.ts

```diff
@@ -233,3 +248,79 @@ export const parameterValues = createReducer(
     );
   },
 );
+
+function newDashboard(
+  before: StoreDashboard,
+  after: Partial<Dashboard>,
+  isDirty: boolean,
+): StoreDashboard {
+  return {
+    ...before,
+    // mimic the StoreDashboard type - this function is only made to update attributes
+    // rather than deep values like dashcards or tabs
+    ...omit(after, "dashcards", "tabs"),
+    embedding_params: syncParametersAndEmbeddingParams(before, after),
+    isDirty,
+  };
+}
+
+export const dashboards = createReducer(
```

Comment:
I couldn't find where this is imported, is this reducer even used?

Reply:
I have that kind of import. It really obscures the code doesn't work well with static code analyzing.

Reply:
yeah, it took more time than I'd like to admit to find it. thankfully we can get rid of it when we migrate more selectors across to TS